ktor-utils/jvm/test/io/ktor/tests/utils/DropLeadingTopDirsTest.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.tests.utils
+
+import io.ktor.util.*
+import kotlin.test.*
+
+class DropLeadingTopDirsTest {
+    @Test
+    fun testEmptyPath() {
+        assertEquals(0, dropLeadingTopDirs(""))
+        assertEquals(0, dropLeadingTopDirs(" "))
+        assertEquals(1, dropLeadingTopDirs("/"))
+        assertEquals(1, dropLeadingTopDirs("\\"))
+    }
+
+    @Test
+    fun testSingleDot() {
+        assertEquals(0, dropLeadingTopDirs("."))
```

Comment:
Dot is just a pointer to the current directory as well as `..` is a pointer to parent. It's not clear why the behavior is different, why we left `.`, but remove `..`.

Reply:
Because `.` is safe and the only dot is not a problem at all. However, parent dir dots should be avoided to get directory "sandbox".

Reply:
Also note that this function is always called after path normalization so we don't need to look deep in a path since `..` could be only at the beginning.

Reply:
The other reason to keep the single dot is that an empty path is questionable while the dot-path is for sure valid.

Reply:
Fixed to remove single dot as well. An empty path result is ok.